### PR TITLE
Br78 test validate

### PR DESCRIPTION
**Pull request by zengxyu**

### CODE REVIEW

File: pytest/validation/test_cross_valildation.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        blueprints = cv.create_blue_prints(blue_print_type=blue_print_type,
+                                           gs_config=gs_config,
+                                           dashify_logging_path=dashify_logging_path,
+                                           num_epochs=num_epochs,
+                                           job_type=job_type)
+        gs_config = GridSearch.create_gs_from_config_dict(gs_config)
+        assert len(blueprints) == len(gs_config) * len(cv._get_fold_indices())
+
+        #
+    # # def test_run(self):
```
Comment (le1nux) [inline pytest/validation/test_cross_valildation.py:110]: commented lines can be deleted

**Review by le1nux — CHANGES_REQUESTED** [review]
Looks good so far. Left a few comments. :)

File: src/ml_gym/util/grid_search.py [hunk 2]
```diff
@@ -221,44 +221,44 @@ def are_sweeps_identical(gs_config_path: str, experiment_folder: str) -> bool:
...
-    for config in configs:
-        print(config)
-
-    print(GridSearch._is_config_equal(configs[0], configs[1]))
-    c = configs[0]
-    c_negligible = {'p_3': {'p_3.1': None}, 'p_1': None, 'p_4': [12, None]}
-    GridSearch._delete_branches(c, c_negligible)
-    print(c)
-    print(GridSearch._is_config_equal(c, c))
+# if __name__ == "__main__":
```
[the remark this answers is not in the document]
Comment (zengxyu) [reply src/ml_gym/util/grid_search.py:224]: done

File: pytest/util/test_grid_search.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        assert GridSearch._is_config_equal(configs[0], configs[0])
+
+        for config in configs:
+            assert GridSearch.is_config_in_gs(config, node)
+        assert not GridSearch.is_config_in_gs({'p_1': 1, 'p_2': 2, 'p_4': [13, 13]}, node)
+
+        c = configs[0]
+        c_negligible = {'p_3': {'p_3.1': None}, 'p_1': None, 'p_4': [12, None]}
+        GridSearch._delete_branches(c, c_negligible)
+        print(GridSearch._is_config_equal(c, c))
```
[the remark this answers is not in the document]
Comment (zengxyu) [reply pytest/util/test_grid_search.py:83]: done

File: pytest/validation/mocked_blueprint.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
+from typing import Dict, List, Any
```
[the remark this answers is not in the document]
Comment (zengxyu) [reply pytest/validation/mocked_blueprint.py:1]: nice idea!

File: pytest/validation/test_cross_valildation.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        indices = cv._get_fold_indices()
+
+        # check that there is no intersection between folds
+        for i, outer_fold_1 in enumerate(indices):
+            for j, outer_fold_2 in enumerate(indices):
+                if i != j:
+                    assert set(outer_fold_1).isdisjoint(outer_fold_2)
+
+        # check stratification
+        folder_indices = np.concatenate(indices, axis=0)
```
[the remark this answers is not in the document]
Comment (zengxyu) [reply pytest/validation/test_cross_valildation.py:84]: I'm not sure if I understand you correctly~ For 5-fold cross-validation, it was split into 5 folds and there is no stratification in each split. I found I wrote the wrong comments here. The indices in each split are shuffled, but I can check the length of each split. :)

File: pytest/validation/test_cross_valildation.py (hunk 1 above)
Comment (zengxyu) [reply pytest/validation/test_cross_valildation.py:110]: done

**Review by le1nux — APPROVED** [review]
LGTM :)
